File: 0932-monotonic-array/0932-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:




        monotone_inc = False
        monotone_dec = False


        for i in range(len(nums)-1):


            # condition for monotonic decresing
            if nums[i] < nums[i+1]:
                monotone_dec = True

            # condition for monotonic incresing
            elif nums[i] > nums[i+1]:
                monotone_inc = True
            
            # if both becomd true means it is not increasing and not decreasing
            if monotone_inc==True and monotone_dec==True:
                return False
        
        return True
        



        

File: 0932-monotonic-array/0932-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {

        boolean monotone_inc = false;
        boolean monotone_dec = false;

        for( int i = 0; i < nums.length -1; i++){

            if (nums[i] < nums[i+1]){
                monotone_inc = true;
            }
            else if( nums[i] > nums[i+1]){
                monotone_dec = true;
            }

            if ((monotone_inc==true) && (monotone_dec == true)){
                return false;
            }

        }
        
        return true;
        
    }
}